### PR TITLE
General Fixes + Format

### PR DESCRIPTION
**Pull request by sc2ad**
This PR provides a wide range of general improvements, mostly centered around configuration and documentation around configuration.

It also supports some formatting + attribute changes, async conversions, default `AsNotTracked` for EF queries, and some other things that I could talk about if commented on.

Hopefully we can discuss merging this in the meeting tomorrow.

### CODE REVIEW

**Review by nike4613 — APPROVED** [review]
LGTM

File: src/CodeGen/Hive.CodeGen/GenericParameterizationGenerator.cs [hunk 1]
```diff
@@ -616,8 +614,11 @@ public TypenameQualifier(SemanticModel model) : base(true)
                 => semModel = model;
 
             public override SyntaxNode? VisitGenericName(GenericNameSyntax node) => QualifyName(node);
```
Comment (sc2ad) [inline src/CodeGen/Hive.CodeGen/GenericParameterizationGenerator.cs:616]: Fix whitespace @sc2ad

File: src/Hive.Permissions/Functions/UserBuiltinFunction.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@ public UserBuiltinFunction(string name, Delegate @delegate)
             this.@delegate = @delegate;
         }
 
+        [SuppressMessage("Design", "CA1031:Do not catch general exception types", Justification = "We catch everything such that we can return false. We log the exception all the same.")]
```
Comment (sc2ad) [inline src/Hive.Permissions/Functions/UserBuiltinFunction.cs:25]: New line

File: src/Hive.Permissions/PermissionActionParseState.cs [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
 using System;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Hive.Permissions
 {
     /// <summary>
     /// A structure that is used to cache parsed actions and related information.
     /// </summary>
     /// <seealso cref="PermissionsManager{TContext}.CanDo(StringView, TContext, ref PermissionActionParseState)"/>
+    [SuppressMessage("Performance", "CA1815:Override equals and operator equals on value types", Justification = "There is no point in overriding this behavior for this type, since we never compare it.")]
```
Comment (sc2ad) [inline src/Hive.Permissions/PermissionActionParseState.cs:12]: New line

File: src/Hive.Permissions/PermissionException.cs [hunk 4]
```diff
@@ -1,11 +1,13 @@
 ﻿using Hive.Utilities;
 using System;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Hive.Permissions
 {
     /// <summary>
     /// An exception that ocurred while compiling or executing rules in the permission system.
     /// </summary>
+    [SuppressMessage("Design", "CA1032:Implement standard exception constructors", Justification = "We are only allowed to make these exceptions around failed permissions")]
```
Comment (sc2ad) [inline src/Hive.Permissions/PermissionException.cs:10]: New line

File: src/Hive.Plugins/Loading/PluginLoaderOptionsBuilder.cs [hunk 5]
```diff
@@ -17,7 +17,9 @@ public class PluginLoaderOptionsBuilder
         internal Action<IConfigurationBuilder, PluginInstance> ConfigurePluginConfigCb = (_, _) => { };
         internal Action<IServiceCollection, PluginInstance> OnPluginLoadedCb = (_, _) => { };
 
-        internal PluginLoaderOptionsBuilder() { }
+        internal PluginLoaderOptionsBuilder()
```
Comment (sc2ad) [inline src/Hive.Plugins/Loading/PluginLoaderOptionsBuilder.cs:20]: Formatting?

**Review by nike4613 — APPROVED** [review]

**Pull request merged**